### PR TITLE
Removed restricted mode - Fix #11972

### LINKED ISSUE #11972: remove -Z ("restricted" mode)
Followup to https://github.com/neovim/neovim/pull/11929#discussion_r386078979 :

`-Z` is a fake-secure mode that behaves slightly differently than "sandbox" mode. It sprinkles more whack-a-mole "secure" code in various places.

Does anyone actually use `-Z`? Where/why is it useful?  

It's fragile to expect applications like N/Vim to plug all potential vectors for eval/shell/etc. For secure scenarios, OS permissions should be used instead.

### PR DESCRIPTION
**Pull request by georg3tom**

### CODE REVIEW

**Comment by bfredl** [comment]
Some linter failures. Please check the  "lint" target of travis or run "make clint" locally.

File: src/nvim/main.c [hunk 1]
```diff
@@ -1023,10 +1023,6 @@ static void command_line_scan(mparm_T *parmp)
           want_argument = true;
           break;
         }
-        case 'Z': {  // "-Z" restricted mode
```
Comment (bfredl) [inline src/nvim/main.c]: Do we want a dedicated error message here (like a reference to `|nvim-features-removed|`) or is the automatic `nvim: Unknown option argument: "-Z"` enough? cc @justinmk

**Comment by georg3tom** [comment]
fixed the linter issues

File: runtime/syntax/msidl.vim [hunk 2]
```diff
@@ -34,7 +34,7 @@ syn region  msidlDefine		start="^[ \t]*#[ \t]*\(define\>\|undef\>\)" skip="\\$"
 
 " Attributes
 syn keyword msidlAttribute      contained in out propget propput propputref retval
-syn keyword msidlAttribute      contained aggregatable appobject binadable coclass control custom default defaultbind defaultcollelem defaultvalue defaultvtable dispinterface displaybind dual entry helpcontext helpfile helpstring helpstringdll hidden id immediatebind lcid library licensed nonbrowsable noncreatable nonextensible oleautomation optional object public readonly requestedit restricted source string uidefault usesgetlasterror vararg version
```
Comment (bfredl) [inline runtime/syntax/msidl.vim]: unrelated, should not be removed.

Comment (georg3tom) [reply runtime/syntax/msidl.vim]: reverted it back

**Pull request merged**